pkg/provider/azure_vmss.go

```diff
@@ -1906,7 +1907,9 @@ func (ss *ScaleSet) ensureBackendPoolDeleted(service *v1.Service, backendPoolIDs
 				return rerr.Error()
 			}
 
+			ss.lockMap.LockEntry(updatedVMLock)
```

Comment:
Can we use atomic.Store() instead of lock?

Reply:
For what purpose? atomic is never used.